File: packages/backend/src/main/java/ru/famsy/backend/modules/user/UserEntity.java
package ru.famsy.backend.modules.user;

import jakarta.persistence.*;

@Entity
public class UserEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    public UserEntity() {}

    public Long getId() {
        return id;
    }

    @Column(nullable = false, unique = true)
    private String username;

    public String getUsername() {
        return username;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    @Column(nullable = false, unique = true)
    private String email;

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    @Column(nullable = false)
    private String password;

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }
}

File: packages/frontend/src/entities/user/entity/user-entity.ts
import { BaseEntity } from '@shared/entity';
import type { UserDTO } from '../dto/user-dto';

export class UserEntity extends BaseEntity {
  public username: string;
  public email: string;

  constructor(dto: UserDTO) {
    super(dto);
    this.username = dto.username;
    this.email = dto.email;
  }
}
